cmd/car/compile.go

```diff
@@ -0,0 +1,446 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"unicode/utf8"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	carv1 "github.com/ipld/go-car"
+	"github.com/ipld/go-car/util"
+	carv2 "github.com/ipld/go-car/v2"
+	"github.com/ipld/go-car/v2/blockstore"
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/codec/dagjson"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	"github.com/ipld/go-ipld-prime/linking"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/basicnode"
+	"github.com/ipld/go-ipld-prime/storage/memstore"
+	"github.com/urfave/cli/v2"
+)
+
+// Compile is a command to translate between a human-debuggable patch-like format and a car file.
+func CompileCar(c *cli.Context) error {
+	var err error
+	inStream := os.Stdin
+	if c.Args().Len() >= 1 {
+		inStream, err = os.Open(c.Args().First())
+		if err != nil {
+			return err
+		}
+	}
+
+	//parse headers.
+	br := bufio.NewReader(inStream)
+	header, _, err := br.ReadLine()
+	if err != nil {
+		return err
+	}
+
+	v2 := strings.Contains(string(header), "--v2")
```

Comment:
probably need to do more than this because the file name could contain this

```suggestion
	v2 := strings.Index(string(header), "car compile --v2 ") != -1
```
?

Reply:
```suggestion
	v2 := strings.HasPrefix(string(header), "car compile --v2 ")
```
that's probably better eh?